server/etcdserver/api/v3rpc/member.go

```diff
@@ -106,7 +106,7 @@ func (cs *ClusterServer) MemberPromote(ctx context.Context, r *pb.MemberPromoteR
 }
 
 func (cs *ClusterServer) header() *pb.ResponseHeader {
-	return &pb.ResponseHeader{ClusterId: uint64(cs.cluster.ID()), MemberId: uint64(cs.server.ID()), RaftTerm: cs.server.Term()}
+	return &pb.ResponseHeader{ClusterId: uint64(cs.cluster.ID()), MemberId: uint64(cs.server.ID()), RaftTerm: cs.server.Term(), Revision: cs.server.KV().Rev()}
```

Comment:
This goes into recent discussion about meaning of revision in lease response requests that was raised during community meeting https://github.com/etcd-io/etcd/issues/13989. Please remember that revision only relates to key value store, even though revision field is present in all etcd rpc responses. Current situation is not satisfactory as usage of revision is inconsistent and not properly documented.

Before we merge this PR we should make a decision of meaning of revision in all rpc responses and document it. We should look into how revision is used currently in APIs, propose changes and analyse impact.

Reply:
I think you mistook `revision` for `raft index`. It's raft index that relates to all data which needs raft concensus. Revision is global version number used for key value store. Etcd implements MVCC by using btree with revision as key. 

If you see this if you look how bbolt is used, we have multiple `buckets` used to stored different values:
* key values are stored under `key` bucket with keys being revisions
* leases are stored under `lease bucket with keys being revision id
* membership info is stored under two buckets `members` and `members_removed` both use member id as key

Reply:
Yes, you are right.  The key for the lease is lease ID. This could be the reason why member related commands do not have revision in the header. 

But all responses share the same header struct, so we might need to hide the revision field if it's 0. For example [printer_fields.go#L39](https://github.com/etcd-io/etcd/blob/main/etcdctl/ctlv3/command/printer_fields.go#L39)

Reply:
+1 to hiding in etcdctl and not exposing revision for these methods.

Reply:
Okay. The missing revision behavior sounds like intended not to reflect unrelated mvcc global revision.